test/general/sycl_iterator/sycl_iterator_definition.pass.cpp

```diff
@@ -0,0 +1,53 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "sycl_iterator_test.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+template <typename _Iterator1, typename _Iterator2, typename = void>
+struct is_equality_implemented : std::false_type
```

Comment:
what is goal of that?
Why do we need that?

Reply:
Ok..
So, I would suggest implement it once as a oneDPL traits "oneapi::dpl::::is_compariable" and implement it depending on C++ version (for C++20 and later just reuse std::equality_comparable)

Reply:
Another variant - if we make decision to check equity of pointers only in this place, this predicate not required anymore.

Reply:
This moment has been fixed in PR https://github.com/oneapi-src/oneDPL/pull/852 and merged into main